service/datastore/client-rest/src/main/java/org/eclipse/kapua/service/datastore/client/rest/RestDatastoreClient.java

```diff
@@ -0,0 +1,641 @@
+        logger.debug("Query - converted query: '{}'", queryMap.toString());
+        long totalCount = 0;
+        Response queryResponse = null;
+        try {
+            queryResponse = esClientProvider.getClient().performRequest(
+                    GET_ACTION,
+                    getSearchPath(typeDescriptor),
+                    Collections.<String, String>emptyMap(),
+                    EntityBuilder.create().setText(MAPPER.writeValueAsString(queryMap)).build(),
+                    new BasicHeader("Content-Type", ContentType.APPLICATION_JSON.toString()));
+            if (isRequestSuccessful(queryResponse)) {
+                JsonNode responseNode = MAPPER.readTree(EntityUtils.toString(queryResponse.getEntity()));
+                JsonNode hitsNode = responseNode.get(KEY_HITS);
+                totalCount = ((NumericNode) hitsNode.get(KEY_TOTAL)).asInt();
+                if (totalCount > Integer.MAX_VALUE) {
+                    throw new ClientException(ClientErrorCodes.ACTION_ERROR, CLIENT_HITS_MAX_VALUE_EXCEDEED);
+                }
+            } else {
+                throw new ClientException(ClientErrorCodes.ACTION_ERROR, queryResponse.getStatusLine().getReasonPhrase());
+            }
+        } catch (ResponseException re) {
+            handleResponseException(re, typeDescriptor.getIndex(), "COUNT");
+        } catch (JsonProcessingException e) {
+            throw new ClientException(ClientErrorCodes.ACTION_ERROR, e, e.getLocalizedMessage());
+        } catch (IOException e) {
+            throw new ClientException(ClientErrorCodes.ACTION_ERROR, e, e.getLocalizedMessage());
+        }
+        return totalCount;
+    }
+
+    @Override
+    public void delete(TypeDescriptor typeDescriptor, String id) throws ClientException {
+        logger.debug("Delete - id: '{}'", id);
+        checkClient();
+        Response deleteResponse = null;
+        try {
+            deleteResponse = esClientProvider.getClient().performRequest(
+                    DELETE_ACTION,
+                    getIdPath(typeDescriptor, id),
+                    Collections.<String, String>emptyMap());
+            if (!isRequestSuccessful(deleteResponse)) {
+                throw new ClientException(ClientErrorCodes.ACTION_ERROR, deleteResponse.getStatusLine().getReasonPhrase());
+            }
+        } catch (ResponseException re) {
+            handleResponseException(re, typeDescriptor.getIndex(), "DELETE");
+        } catch (IOException e) {
+            throw new ClientException(ClientErrorCodes.ACTION_ERROR, e, e.getLocalizedMessage());
+        }
+    }
+
+    @Override
+    public void deleteByQuery(TypeDescriptor typeDescriptor, Object query) throws ClientException {
+        checkClient();
+        JsonNode queryMap = queryConverter.convertQuery(query);
+        JsonNode deleteRequestNode = queryMap.get(SchemaKeys.KEY_QUERY);
+        logger.debug("Query - converted query: '{}'", queryMap.toString());
```

Comment:
Remove `.toString()` is already called `.toString()` for each parameter passed as argument.